src/logic/function/iterate.lean

```diff
@@ -149,4 +149,7 @@ theorem right_inverse.iterate {g : α → α} (hg : right_inverse g f) (n : ℕ)
   right_inverse (g^[n]) (f^[n]) :=
 hg.iterate n
 
+lemma iterate_comm {m n : ℕ} : f^[n]^[m] = (f^[m]^[n] : α → α) :=
```

Comment:
A possibly outrageous idea:
```suggestion
lemma iterate_comm (m n : ℕ) : function.commute (iterate m : (α → α) → (α → α)) (iterate n) :=
```

Reply:
Eric, I do not understand: what is `iterate m` in your lemma?  I tried copying it, and Lean says
```
unknown identifier 'iterate'
```

Reply:
That tells me your lemma is in the wrong namespace! `^[m]` is `nat.iterate m`.

Reply:
At any rate, the statement was nonsense, what I meant was
```lean
lemma iterate_commute (m n : ℕ) : function.commute (λ f : α → α, f^[m]) (λ f, f^[n])
```
which is probably worth having in addition to your lemma, rather than instead of it.

Reply:
It's `nat.iterate`, the function behind the `f^[n]` notation.

Reply:
Eric, this works:
```lean
lemma iterate_commute {α : Type*} (m n : ℕ) : function.commute (λ f : α → α, f^[m]) (λ f, f^[n]) :=
λ f, function.iterate_comm
```
Is this a better way of formulating the lemma?

Reply:
Ok, I changed the lemma statement and pushed it!

Reply:
No, I meant literally define `commute` in terms of comm - I think having both is useful!

Reply:
Ok, both variants are in the PR simultaneously now!

Reply:
Having both is useful because `h := function.iterate_commute m n f` returns an equality with `λ f, f^[m]`.

Reply:
Thank you both for the suggestions and explanations!